frontend/packages/ux-editor/src/components/config/editModal/EditImage/LocalImage/ImportImage/ImportImage.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useRef, useState } from 'react';
+import { StudioButton } from '@studio/components';
+import classes from './ImportImage.module.css';
+import { useAddImageMutation } from 'app-shared/hooks/mutations/useAddImageMutation';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { AddImageFromLibraryModal } from './AddImageFromLibrary/AddImageFromLibraryModal';
+import { UploadImage } from './UploadImage/UploadImage';
+import { useTranslation } from 'react-i18next';
+import type { AxiosError } from 'axios';
+import type { ApiError } from 'app-shared/types/api/ApiError';
+import { WWWROOT_FILE_PATH } from '../../constants';
+import type { UseMutateFunction } from '@tanstack/react-query';
+import type { TFunction } from 'i18next';
+
+interface ImportImageProps {
+  onImageChange: (imageSource: string) => void;
+}
+
+const CONFLICTING_IMAGE_FILE_NAME_API_ERROR_CODE = 'AD_04';
+
+export const ImportImage = ({ onImageChange }: ImportImageProps) => {
+  const { t } = useTranslation();
+  const [showChooseFromLibraryModalOpen, setShowChooseFromLibraryModalOpen] =
+    useState<boolean>(false);
+  const imageUploaderRef = useRef(null);
+  const { org, app } = useStudioEnvironmentParams();
+  const { mutate: uploadImage } = useAddImageMutation(org, app, true);
+
+  const handleSubmit = async (event?: FormEvent<HTMLFormElement>) => {
+    event?.preventDefault();
+    const imageFile = getImageFile(imageUploaderRef);
+    uploadImageWithDuplicateHandling(imageFile, uploadImage, onImageChange, t);
+  };
+
+  const handleInputChange = async () => {
+    const file = getImageFile(imageUploaderRef);
+    if (file) await handleSubmit();
+  };
+
+  return (
+    <div className={classes.importImage}>
+      <StudioButton size='small' onClick={() => setShowChooseFromLibraryModalOpen(true)}>
+        {t('ux_editor.properties_panel.images.choose_from_library')}
+      </StudioButton>
+      {showChooseFromLibraryModalOpen && (
+        <AddImageFromLibraryModal
+          isOpen={showChooseFromLibraryModalOpen}
+          onClose={() => setShowChooseFromLibraryModalOpen(false)}
+          onAddImageReference={onImageChange}
+        />
+      )}
+      <UploadImage
+        onHandleSubmit={handleSubmit}
+        onHandleInputChange={handleInputChange}
+        imageRef={imageUploaderRef}
+      />
+    </div>
+  );
+};
+
+const getImageFile = (imageUploaderRef: MutableRefObject<any>) =>
+  imageUploaderRef?.current?.files?.item(0);
+
+const uploadImageWithDuplicateHandling = (
+  imageFile,
+  uploadImage: UseMutateFunction<FormData, Error, FormData, unknown>,
+  onImageChange: (imageSource: string) => void,
+  t: TFunction,
+) => {
+  if (imageFile) {
+    const formData = new FormData();
+    formData.append('image', imageFile);
+    uploadImage(formData, {
+      onError: (error: AxiosError<ApiError>) =>
+        error.response.data.errorCode === CONFLICTING_IMAGE_FILE_NAME_API_ERROR_CODE &&
```

Comment:
Det er ikke helt heldig at vi baserer et tiltenkt scenario på at vi får en feilmelding fra endepunktet. Backendkallene skal i utgangspunktet bare feile når det er en feil fra vår side (f.eks. i infrastruktur eller kode). Har du vurdert andre måter å sjekke for duplikater på? Kunne vi hatt et eget endepunkt som gjør det? Eller har vi allerede filnavnene tilgjengelige et sted i mellomlageret, så vi kan sjekke mot det?